shared/react-native/ios/KeybaseShare/ShareViewController.m

```diff
@@ -329,6 +336,74 @@ - (void)processItem:(NSItemProvider*)item lastItem:(BOOL)lastItem {
   }
 }
 
+// processFilesItem will upload the shared file to the Keybase filesystem.
+- (void)processFilesItem:(NSItemProvider*)item lastItem:(BOOL)lastItem {
+  NSString* path = self.folderTarget;
+  NSItemProviderCompletionHandler urlHandler = ^(NSURL* url, NSError* error) {
+    [[[PushNotifier alloc] init] localNotification:@"extension" msg:@"You cannot share a URL to a file."
+                                        badgeCount:-1 soundName:@"default" convID:@"" typ:@"chat.extension"];
+    [self maybeCompleteRequest:lastItem];
+  };
+  
+  NSItemProviderCompletionHandler textHandler = ^(NSString* text, NSError* error) {
+    // Create a text file out of the URL.
+    // KeybaseExtensionSaveTextFile(path, url, self.contentText, &error);
+    [[[PushNotifier alloc] init] localNotification:@"extension" msg:@"Sharing text not yet implemented."
+                                        badgeCount:-1 soundName:@"default" convID:@"" typ:@"chat.extension"];
+    [self maybeCompleteRequest:lastItem];
+  };
+  
+  // If we get an image in the form of a UIImage, we first write it to a temp file
+  // and run it through the normal file sharing code path.
+  NSItemProviderCompletionHandler imageHandler = ^(UIImage* image, NSError* error) {
+    NSString* guid = [[NSProcessInfo processInfo] globallyUniqueString];
+    NSString* filename = [NSString stringWithFormat:@"%@%@.jpg", NSTemporaryDirectory(), guid];
+    NSURL* url = [NSURL fileURLWithPath:filename];
+    
+    NSData* imageData = UIImageJPEGRepresentation(image, 0.7);
```

Comment:
I know chat does the same, but am curious if 0.7 is the JPEG quality and if we should have picked a higher value.

Reply:
I'm just going to leave this in the interest of minimizing file size. The quality seems more than good enough.